cmd/connections.go

```diff
@@ -465,3 +475,636 @@ func terminateConnectionByUser(options connectionOptions) error {
+}
+
+//
+//
+func uebaAnomalySettingsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "anomaly-settings",
+		Short: "Get and create Ueba Anomaly Settings",
+		Long:  `Get and create Ueba Anomaly Settings`,
+		Example: `
+	privx-cli ueba anomaly-settings [access flags]
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return uebaAnomalySettings(args)
+		},
+	}
+
+	cmd.AddCommand(uebanomalySettingsCreateCmd())
+
+	return cmd
+}
+
+func uebaAnomalySettings(args []string) error {
+	api := connectionmanager.New(curl())
+
+	settings, err := api.UebaAnomalySettings()
+	if err != nil {
+		return err
+	}
+
+	return stdout(settings)
+}
+
+//
+//
+func uebanomalySettingsCreateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create Ueba Anomaly Settings",
+		Long:  `Create Ueba Anomaly Settings`,
+		Example: `
+	privx-cli ueba anomaly-settings create [access flags] JSON-FILE
+		`,
+		Args:         cobra.ExactArgs(1),
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return uebaAnomalySettingsCreate(args)
+		},
+	}
+
+	return cmd
+}
+
+func uebaAnomalySettingsCreate(args []string) error {
+	var newSettings connectionmanager.UebaAnomalySettings
+	api := connectionmanager.New(curl())
+	err := decodeJSON(args[0], &newSettings)
+	if err != nil {
+		return err
+	}
+
+	err = api.CreateAnomalySettings(newSettings)
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+//
+//
+func uebaStartAnalyzingCmd() *cobra.Command {
+	options := uebaOptions{}
+	cmd := &cobra.Command{
+		Use:   "start-analysis",
+		Short: "Start Ueba analyzing",
+		Long:  `Start analyzing connections with a saved dataset. Fails if training not done, has not finished or failed.`,
+		Example: `
+	privx-cli ueba start-analysis [access flags] --id <DATASET-ID>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return uebaStartAnalyzing(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.datasetID, "id", "", "dataset ID")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func uebaStartAnalyzing(options uebaOptions) error {
+	api := connectionmanager.New(curl())
+	err := api.StartAnalyzing(options.datasetID)
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+//
+//
+func uebaStopAnalyzingCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "stop-analysis",
+		Short: "Stop Ueba analyzing",
+		Long:  `Stop analyzing connections with a saved dataset. Fails if training not done, has not finished or failed.`,
+		Example: `
+	privx-cli ueba stop-analysis [access flags]
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return uebaStopAnalyzing()
+		},
+	}
+
+	return cmd
+}
+
+func uebaStopAnalyzing() error {
+	api := connectionmanager.New(curl())
+	err := api.StopAnalyzing()
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+//
+//
+func uebaSetupScriptCmd() *cobra.Command {
```

Comment:
From user point of view both sub-commands need to be combined to produce a single download action. In the back-end it is divided into 2 steps: getting the download handle (create download session id) and then using the handle to download the script in a separate GET request.

privx-cli would just combine them together with an option to provide alternative destination path.

Reply:
ok. didnt know about that. I just leave a single download command then.